### PR TITLE
SOM: check if data has single attribute

### LINKED ISSUE #4126: OWSOM: crash on single column
**Describe the bug**
SOM crashes when given data with a single column.

**To Reproduce**
1. File (iris)
2. Select Columns (remove all but one attribute=
3. Connect to SOM.

**Expected behavior**
A warning appears.

### PR DESCRIPTION
**Pull request by ajdapretnar**
##### Issue

Fixes #4126. 

##### Description of changes
Check if data has single attribute.

##### Includes
- [X] Code changes
- [X] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by janezd** [comment]
There is no reason why SOM wouldn't work with a single variable. (It doesn't make sense to use SOM then, but this is not a reason. :)

#4126 happens because initial weights cannot be initialized with PCA when the data has just a single dimension. If you switching to random initialization, SOM works even with a single attribute.

The proper fix, I think, is to switch to random initialization in this case. We already do the same for sparse data.

```
--- a/Orange/projection/som.py
+++ b/Orange/projection/som.py
@@ -50,7 +50,7 @@ class SOM:
                 f(self.weights, x,
                   sigma / decay, learning_rate / decay)

-        if self.pca_init and not sp.issparse(x):
+        if self.pca_init and not sp.issparse(x) and x.shape[1] > 1:
             self.init_weights_pca(x)
         else:
             self.init_weights_random(x)
```

File: Orange/widgets/unsupervised/owsom.py [hunk 1]
```diff
@@ -338,6 +339,8 @@ def set_warnings():
             cont_attrs = [var for var in attrs if var.is_continuous]
             if not cont_attrs:
                 self.Error.no_numeric_variables()
+            elif len(attrs) == 1:
+                self.Error.single_attribute()
```
Comment (janezd) [inline Orange/widgets/unsupervised/owsom.py]: But if we follow this route, check `len(cont_attrs)` not `len(attrs)`.

Comment (ajdapretnar) [reply Orange/widgets/unsupervised/owsom.py]: No, I prefer your approach. We are not to block people from doing silly things. :)

**Comment by ajdapretnar** [comment]
I decided for a warning and checking the attributes at projection level.

File: Orange/widgets/unsupervised/owsom.py (hunk 1 above)
Comment (janezd) [reply Orange/widgets/unsupervised/owsom.py]: OK, but you must check `cont_attrs`, not `attrs` (and change the message from `(...) single column` to `(...) single numeric variable`). It doesn't help if the data has 1000 categorical features but a single numeric.

Comment (ajdapretnar) [reply Orange/widgets/unsupervised/owsom.py]: It was before my coffee. I was sure now I am checking for cont_attrs.

**Pull request merged**